src/commands/api/findTreeItem.ts

```diff
@@ -114,3 +114,44 @@ async function searchDbAccounts(dbAccounts: AzExtTreeItem[], expected: ParsedCon
 
     return undefined;
 }
+
+async function searchCosmosDbAccount(dbAccount: MongoAccountTreeItem | DocDBAccountTreeItemBase, expected: ParsedConnectionString, actual: ParsedConnectionString, context: IActionContext): Promise<DatabaseAccountTreeItem | DatabaseTreeItem | undefined> {
+
+    if (expected.accountId === actual.accountId) {
+        if (expected.databaseName) {
+            const dbs = await dbAccount.getCachedChildren(context);
+            for (const db of dbs) {
+                if ((db instanceof MongoDatabaseTreeItem || db instanceof DocDBDatabaseTreeItemBase) && expected.databaseName === db.databaseName) {
+                    return new DatabaseTreeItemInternal(expected, expected.databaseName, dbAccount, db);
+                }
+            }
+            // We found the right account - just not the db. In this case we can still 'reveal' the account
+            return new DatabaseTreeItemInternal(expected, expected.databaseName, dbAccount);
+        }
+        return new DatabaseAccountTreeItemInternal(expected, dbAccount);
+    }
+
+    return undefined;
+}
+
+async function searchPostgresServerAccount(dbAccount: PostgresServerTreeItem, expected: ParsedConnectionString, actual: ParsedConnectionString, context: IActionContext): Promise<DatabaseAccountTreeItem | DatabaseTreeItem | undefined> {
```

Comment:
> Code Changes to findTreeItem.ts to fix funtionality issues

Can you explain what the functionality issues were? I'm just not sure why we need a separate "searchPostgresServerAccount"

Reply:
The current search function has an odd structure, where when the `database` is not found in the attachedAccounts list however if the `account` is found, it will immediately return a value rather than undefined. 
`// We found the right account - just not the db. In this case we can still 'reveal' the account`
`return new DatabaseTreeItemInternal(expected, expected.databaseName, dbAccount);`
With undefined returned the call will move down to searching azureSubscriptions instead and usually the right account is found in this step. 

We can have one single function for all but it requires changing existing code.

Reply:
Remember the attached scenario isn't really for Azure users. We don't expect someone to attach an Azure server when they can already see it in their subscriptions (I know you're doing it, but that's really only for testing purposes). So it's actually unlikely that "the right account is found" by searching the azure subscriptions.

Reply:
That's true I forgot about that. In that case sticking with the old format for postgres as well should work.